render-app/src/main/java/org/janelia/alignment/destreak/ConfigurableMaskStreakExtractor.java

```diff
@@ -0,0 +1,63 @@
+package org.janelia.alignment.destreak;
+
+import ij.ImagePlus;
+import ij.process.ImageProcessor;
+
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.converter.Converters;
+import net.imglib2.img.Img;
+import net.imglib2.img.display.imagej.ImageJFunctions;
+import net.imglib2.type.numeric.integer.UnsignedByteType;
+import net.imglib2.type.numeric.real.FloatType;
+
+/**
+ * Extension of {@link ConfigurableMaskStreakCorrector} filter that produces an image containing
+ * extracted streaks instead of a de-streaked image.  This implementation was created to support
+ * analysis of Christopher Bleck's tough resin experiments.
+ * The streak extraction process is very slow and involves multiple image copies and conversions,
+ * therefore this filter should primarily be used for distributed client-side processing
+ * (as opposed to using it for dynamic server-side processing).
+ */
+public class ConfigurableMaskStreakExtractor
+        extends ConfigurableMaskStreakCorrector {
+
+    public ConfigurableMaskStreakExtractor() {
+    }
+
+    public ConfigurableMaskStreakExtractor(final ConfigurableMaskStreakCorrector corrector) {
+        super(corrector);
+    }
+
+    @Override
+    public void process(final ImageProcessor ip,
+                        final double scale) {
+
+        // save original image for later subtraction
+        final ImagePlus originalImagePlus = new ImagePlus("original", ip.duplicate());
+        final Img<UnsignedByteType> original = ImageJFunctions.wrapByte(originalImagePlus);
```

Comment:
I guess that is just a stylistic difference between us.  I usually prefer not to have so much inlined because it takes me longer to unpack what is happening.  I also like having only one call per line because it simplifies debugging later when things go wrong.  In this case the extra "complexity" is created by the conversions back and forth between ImageJ1 and ImageJ2 stuff.